src/core/execute.c

```diff
@@ -4181,6 +4189,28 @@ void exec_context_dump(const ExecContext *c, FILE* f, const char *prefix) {
                                 c->bind_mounts[i].recursive ? "rbind" : "norbind");
                 }
 
+        if (c->n_temporary_filesystems > 0)
+                for (i = 0; i < c->n_temporary_filesystems; i++) {
+                        _cleanup_free_ char *str = NULL;
+                        TemporaryFileSystem *t = c->temporary_filesystems + i;
+                        unsigned long flags;
+
+                        str = strdup(strempty(t->options));
+                        if (!str)
+                                continue;
+
+                        flags = t->flags | (t->read_only ? MS_RDONLY : 0);
+
+                        r = mnt_optstr_apply_flags(&str, flags, mnt_get_builtin_optmap(MNT_LINUX_MAP));
+                        if (r < 0)
+                                continue;
+
+                        fprintf(f, "%sTemporaryFileSystem: %s%s:%s\n", prefix,
+                                t->ignore_enoent ? "-" : "",
+                                t->path,
+                                str);
```

Comment:
hmm, for the dumping I'd probably just output the options string as it is, i.e. not process it with mnt_optstr_apply_flags or such

Reply:
OK. Updated.